x-pack/plugins/secops/public/components/timeline/search_or_filter/search_or_filter.tsx

```diff
@@ -85,22 +95,42 @@ export const SearchOrFilter = pure<Props>(
         </ModeFlexItem>
         <EuiFlexItem data-test-subj="timeline-search-container">
           <EuiToolTip content={modes[kqlMode].kqlBarTooltip}>
-            <EuiSearchBar
-              data-test-subj="timeline-search-bar"
-              box={{
-                placeholder: modes[kqlMode].placeholder,
-                incremental: false,
-                schema: {
-                  flags: [],
-                  fields: {},
-                },
-              }}
-              query={kqlQuery}
-              onChange={({ queryText }: { queryText: string }) =>
-                // TODO: this handler is NOT being called by `EuiSearchBar`, which causes the query entered by the user to disappear
-                updateKqlQuery({ id: timelineId, kqlQuery: queryText })
-              }
-            />
+            <>
+              {kqlMode === 'filter' && (
+                <EuiSearchBar
+                  data-test-subj="timeline-search-bar"
+                  box={{
+                    placeholder: modes[kqlMode].placeholder,
+                    incremental: false,
+                    schema: {
+                      flags: [],
+                      fields: {},
+                    },
+                  }}
+                  query=""
+                  onChange={({ queryText }: { queryText: string }) =>
+                    // TODO: this handler is NOT being called by `EuiSearchBar`, which causes the query entered by the user to disappear
+                    alert('I will be doing filter at one point')
```

Comment:
Noted, you have an alert but since it's not being called it is ok to leave it here. Just pointing it out in case you want to remove it before merging this all in.

Reply:
it will be called at one point I will move it too console.log it was there for testing purpose